src/hotspot/share/opto/loopTransform.cpp

```diff
@@ -4381,6 +4381,19 @@ bool PhaseIdealLoop::intrinsify_fill(IdealLoopTree* lpt) {
   Node* len = new SubINode(head->limit(), head->init_trip());
   _igvn.register_new_node_with_optimizer(len);
 
+  // If the store is on the backedge, it is not executed in the last
+  // iteration, and we must subtract 1 from the len.
+  Node* backedge = head->loopexit()->proj_out(1);
+  if (store->in(0) == backedge) {
+    len = new SubINode(len, _igvn.intcon(1));
+    _igvn.register_new_node_with_optimizer(len);
```

Comment:
How about moving the earlier call to register_new_node_with_optimizer to after this block, so it is only called once?

Reply:
Hi @dean-long . Just discussed it with @TobiHartmann . We have two new nodes here, one to compute
`len = limit - init`,
the other to decrement the len by 1:
`len = len - 1`.
Both are new nodes, and need to be registered with igvn. So we think it is correct the way it is now.

Reply:
OK, that makes sense.